### ISSUE TITLE
Unit tests fail with can't modify frozen String (RuntimeError)

### ISSUE THREAD

**Issue by edergillian** [puppetlabs/pdk#886]
**Describe the bug**
When trying to run a unit test using `pdk test unit --tests <path>` on Windows, I get this error:
```
C:/Program Files/Puppet Labs/DevelopmentKit/private/ruby/2.4.10/lib/ruby/gems/2.4.0/gems/pdk-1.18.0/lib/pdk/tests/unit.rb:96:in `tr!': can't modify frozen String (RuntimeError)
        from C:/Program Files/Puppet Labs/DevelopmentKit/private/ruby/2.4.10/lib/ruby/gems/2.4.0/gems/pdk-1.18.0/lib/pdk/tests/unit.rb:96:in `invoke'
        from C:/Program Files/Puppet Labs/DevelopmentKit/private/ruby/2.4.10/lib/ruby/gems/2.4.0/gems/pdk-1.18.0/lib/pdk/cli/test/unit.rb:80:in `block (2 levels) in <module:CLI>'
        from C:/Program Files/Puppet Labs/DevelopmentKit/private/ruby/2.4.10/lib/ruby/gems/2.4.0/gems/cri-2.10.1/lib/cri/command.rb:329:in `run_this'
        from C:/Program Files/Puppet Labs/DevelopmentKit/private/ruby/2.4.10/lib/ruby/gems/2.4.0/gems/cri-2.10.1/lib/cri/command.rb:269:in `run'
        from C:/Program Files/Puppet Labs/DevelopmentKit/private/ruby/2.4.10/lib/ruby/gems/2.4.0/gems/cri-2.10.1/lib/cri/command.rb:287:in `run'
        from C:/Program Files/Puppet Labs/DevelopmentKit/private/ruby/2.4.10/lib/ruby/gems/2.4.0/gems/cri-2.10.1/lib/cri/command.rb:287:in `run'
        from C:/Program Files/Puppet Labs/DevelopmentKit/private/ruby/2.4.10/lib/ruby/gems/2.4.0/gems/pdk-1.18.0/lib/pdk/cli.rb:68:in `run'
        from C:/Program Files/Puppet Labs/DevelopmentKit/private/ruby/2.4.10/lib/ruby/gems/2.4.0/gems/pdk-1.18.0/exe/pdk:6:in `<top (required)>'
        from C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.4.10/bin/pdk:23:in `load'
        from C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.4.10/bin/pdk:23:in `<main>'
```

It didn't happen on PDK 1.14.1. It started happening after I upgraded to 1.18.0. I updated the module to PDK 1.18 template before running the tests, but that didn't affect the tests already written (no magic comment on frozen_string_literal on them). But even including the comment, the error persists.

It seems to me that it happens when PDK tries to convert Windows path to Unix path, in units.rb, line 96:
`tests.tr!('\\', '/') unless tests.nil?`

Obviously, it does not error out if run without the `tests` parameter.

As a workaround, I commented out that line and it's working again.

**To Reproduce**
Install PDK 1.18.0
Run a unit test on a module written in PDK 1.14.1 and updated to 1.18 template, using the `tests` parameter, like:
`pdk test unit --tests <path>`

The path does not need to be Windows-like, it just has to exist.

**Expected behavior**
It should run the tests normally.

**Additional context**
- PDK installation method: tried with both MSI and chocolatey.
- PDK version: 1.18.0
- Windows 10 Professional

**Comment by natemccurdy** [comment]
Seeing this same problem when running `pdk test unit --tests path/to/spec.rb` on MacOS

PDK version 1.18.0

**Comment by natemccurdy** [comment]
I've fixed this locally by changing [line 96](https://github.com/puppetlabs/pdk/blob/v1.18.0/lib/pdk/tests/unit.rb#L96) to:
```ruby
tests = tests.tr('\\', '/') unless tests.nil?
```

**Comment by natemccurdy** [comment]
This has been fixed in #891
